compiler/ml/typecore.ml

```diff
@@ -2637,7 +2651,7 @@ and type_expect_ ?type_clash_context ?in_function ?(recarg = Rejected) env sexp
       }
   | Pexp_record (lid_sexp_list, Some sexp) ->
     assert (lid_sexp_list <> []);
-    let exp = type_exp ~recarg env sexp in
+    let exp = type_exp ~context:None ~recarg env sexp in
```

Comment:
@zth this is where I believe one can pass context to special case error messages for e.g. optional fields.